test/book/specialized/brandhorst-zach-fibration-hopping/vinberg_2.jlcon

```diff
@@ -100,19 +119,19 @@ julia> @assert gram_matrix(NSY1) == gram_matrix(NS)[I,I]
 julia> Oscar.horizontal_decomposition(Y1, fibers[2][I])[2]
 Effective weil divisor
   on elliptic surface with generic fiber -x^3 + y^2 - t^7 + 2*t^6 - t^5
-with coefficients in integer Ring
+with coefficients in integer ring
 given as the formal sum of
+  2 * component E8_3 of fiber over (0, 1)
```

Comment:
If the order here keeps changing, that suggests a missing `hash` method for somewhere; well, most likely for the keys (resp. the type of the keys) in some dictionary. Is there a dict being used here? What is the type of the keys?

Reply:
The type of the dictionary is
`IdDict{Oscar.AbsIdealSheaf, ZZRingElem}`

Reply:
so `AbsIdealSheaf` is an abstract type, but all its subtypes seem to end with `IdealSheaf`. Alas `git grep hash.*IdealSheaf` reveals zero hits, i.e. no hashing methods for any of those types. At the same time, there *is* an equality method in `experimental/Schemes/src/IdealSheaves.jl:466`:
```julia
function ==(I::AbsIdealSheaf, J::AbsIdealSheaf)
  I === J && return true
  X = space(I)
  X == space(J) || return false
  for U in basic_patches(default_covering(X))
    is_subset(I(U), J(U)) && is_subset(J(U), I(U)) || return false
  end
  return true
end
```

But the rules are clear: when you define a custom `==` method you have to ensure `hash` still satisfies the implication `if a==b then hash(a)==hash(b)` which usually means a custom hash method has to be provided.


The absolute minimal "always correct" hash function would be this:
```julia
Base.hash(x::AbsIdealSheaf, h::UInt) = h
```
So you could try adding this, perhaps with a comment `# TODO: replace by something better`

The drawback of such a hash function is that it makes lookups of a key in a Dict slower. The upside is that it also makes it less wrong, which I think is overall preferable in our business... ;-)

Reply:
Aha, but you are actually storing things in an `IdDict` -- so in that case the `hash` method does not matter at all.

But also: you have zero control over the order, because now instead of `hash` it uses `objectid` which essentially is the address of the object, and that of course can change when you re-run the program.


In that case the only fix I can think of is to modify your printing code to "sort" the keys in some way, based on some parameters -- doesn't have to be perfect (i.e. not a total order), just good enough to stabilize the output for this example.

Reply:
In this example, perhaps one could "cheaply sort" the entries so that those with a "component" in the output string come (for example) first; and then those in turn are sorted on these strings `E8_4` etc.; and when there is a tie, sort next by the points they are a fiber over (i.e. `(0,1)` and so on?

Reply:
(Or switch to a `Dict` after adding a suitable `hash` function -- but perhaps your `==` method is too slow for that?)

BTWH `hash(h) = h` will still not give fully stable output, but assuming your algorithm is deterministic, or at least runs with a fixed RNG seed (as is the case here) the order should be stable.

Reply:
As you guessed we chose `IdDict` because `==` is prohibitively expensive